file/file.go

```diff
@@ -187,17 +195,21 @@ func NewWriter(pth string, opt *Options) (w Writer, err error) {
 		opt = NewOptions()
 	}
 
-	switch parseScheme(pth) {
+	u, err := url.Parse(pth)
+	if err != nil {
+		return
+	}
+	bufOpts := bufOptions(*opt)
+	switch u.Scheme {
 	case "s3":
-		accessKey := opt.AccessKey
-		secretKey := opt.SecretKey
-		s3Opts := s3Options(*opt)
-		w, err = s3.NewWriter(pth, accessKey, secretKey, &s3Opts)
+		return minio.NewWriter(pth, awsHost, opt.AccessKey, opt.SecretKey, &bufOpts)
 	case "gcs", "gs":
 		accessKey := opt.AccessKey
 		secretKey := opt.SecretKey
 		gcsOpts := gcsOptions(*opt)
 		w, err = gs.NewWriter(pth, accessKey, secretKey, &gcsOpts)
+	case "m3", "minio":
```

Comment:
@jbsmith7741  it looks like you allow a scheme of `m3://` or `minio://` but below in your minio/read_test.go examples it has `ms://`  It looks like the tests are passing, so I'm not sure how this works?

Reply:
I've updated them to use `mc` which is properly the best to use as minio has a tool called mc that works like gcutils and s3